### DISCUSSION TITLE
`rootDir` configuration (to potentially support mono repos)

### DISCUSSION THREAD

**Discussion by lukethacoder** [forcedotcom/salesforcedx-vscode#5474] [Ideas]
**Is your feature request related to a problem? Please describe.**

Attempting to use this extension in a monorepo does not work as expected due to the plugin expecting the SFDX Project to be the root directory.

**Describe the solution you'd like**

Add a `rootDir` plugin config option to flag which folder should be used as the SFDX project root

**Describe alternatives you've considered**

Can open the individual folder in different VS Code instance, but I would rather stay within one instance of the mono repo.

**Additional context**

**Comment by jasonAtThinkLP** [comment]
This really feels like it should be both extremely simple to implement, and like table stakes for an extension set like this. 

Why should a VS Code extension have any say whatsoever about how I structure my repo and project and documentation files? It's pretty common to build an app that consists of multiple applications and services all nested within a single repo.

**Comment by nwcm** [comment]
This would also be a good addition and basic functionality. If you just need a mono repo of multiple SFDX packages you can achieve this with `/src/packageA/` and sfdx-project in root. However, many of the lower extensions, typing, autocomplete (language server) and such will basically be useless to you
